crates/resource-resolver/build.rs

```diff
@@ -0,0 +1,13 @@
+use std::env;
+
+fn main() {
+    track_var("CARGO_PACKAGER_FORMAT");
+    track_var("CARGO_PACKAGER_MAIN_BINARY_NAME");
+}
+
+fn track_var(key: &str) {
+    println!("cargo:rerun-if-env-changed={}", key);
+    if let Ok(var) = env::var(key) {
+        println!("cargo:rustc-cfg={}=\"{}\"", key, var);
+    }
+}
```

Comment:
let's simplify this since we only need to track `CARGO_PACKAGER_FORMAT`, also this tracking should be done only if `auto-detect-format` feature is active.